src/cargo/core/package_id.rs

```diff
@@ -10,6 +11,7 @@ use std::sync::OnceLock;
 use serde::de;
 use serde::ser;
 
+use crate::core::resolver::activation_key::ActivationKey;
```

Comment:
If the `ActivationKey` is going to be part of a fundamental data structure like `PackageId` then it should live in the same area, not in the resolver. @epage will probably have opinions about where, but I would guess `crate::core::ActivationKey;`.

Reply:
Cargo's subsystems are a bit of a mess right now and I'd love to go through and make them more strictly hierarchical.  We have other "violations" right now so I'm not too sure if changing this will make much of a differene.

I do think I'd want us to have a clearer name within `package_id.rs`